File: java/src/com/zhy/DesignPattern/Prototype/Deep/Student.java
package com.zhy.DesignPattern.Prototype.Deep;

import com.zhy.DesignPattern.Prototype.AbstractStudent;
import com.zhy.DesignPattern.Prototype.Course;

public class Student extends AbstractStudent {

	@Override
	public AbstractStudent Clone() {
		
		Student student = new Student();
		student.setName(super.getName());
		student.setAge(super.getAge());
		student.setCourse(new Course());
		student.getCourse().setName(super.getCourse().getName());
		return student;
	}
}

File: csharp/DesignPattern.Prototype/Deep/Student.cs

namespace DesignPattern.Prototype.Deep
{
    /// <summary>
    /// 深拷贝
    /// </summary>
    public class Student : AbstractStudent
    {
        public override AbstractStudent Clone()
        {
            Student student = new Student();
            student.Name = base.Name;
            student.Age = base.Age;
            student.Course = new Course();
            student.Course.Name = base.Course.Name;

            return student;
        }   
    }
}
